### ISSUE TITLE
!unsafe tags not supported

### ISSUE THREAD

**Issue by rhpvorderman** [willthames/ansible-lint#291]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible --version = 2.3.0.0
ansible-lint --version 3.4.16
```

- ansible installation method: pip
- ansible-lint installation method: pip
# Desired Behaviour

Ansible-lint works on a playbook that has defined unsafe variables with !unsafe.

# Actual Behaviour (Bug report only)

An error occurs when running ansible-lint on the following case:

In this case, a docker inspect command is run which also uses double bracket notations. Therefore these need to be escaped with the !unsafe tag.

```YAML
- name: register whether the container has finished
  command: "docker inspect -f {{inspect}} {{container_name}}"
  vars:
    inspect: !unsafe '{{.State.Running}}'
  register: container_running
```

Output:
```
Traceback (most recent call last):
  File "/project_folder/.venv/bin/ansible-lint", line 11, in <module>
    sys.exit(main())
  File "/project_folder/.venv/local/lib/python2.7/site-packages/ansiblelint/__main__.py", line 122, in main
    matches.extend(runner.run())
  File "/project_folder/.venv/local/lib/python2.7/site-packages/ansiblelint/__init__.py", line 243, in run
    skip_list=set(self.skip_list)))
  File "/project_folder/.venv/local/lib/python2.7/site-packages/ansiblelint/__init__.py", line 139, in run
    matches.extend(rule.matchtasks(playbookfile, text))
  File "/project_folder/.venv/local/lib/python2.7/site-packages/ansiblelint/__init__.py", line 66, in matchtasks
    yaml = ansiblelint.utils.parse_yaml_linenumbers(text, file['path'])
  File "/project_folder/.venv/local/lib/python2.7/site-packages/ansiblelint/utils.py", line 527, in parse_yaml_linenumbers
    data = loader.get_single_data()
  File "/project_folder/.venv/local/lib/python2.7/site-packages/yaml/constructor.py", line 39, in get_single_data
    return self.construct_document(node)
  File "/project_folder/.venv/local/lib/python2.7/site-packages/yaml/constructor.py", line 48, in construct_document
    for dummy in generator:
  File "/project_folder/.venv/local/lib/python2.7/site-packages/yaml/constructor.py", line 398, in construct_yaml_map
    value = self.construct_mapping(node)
  File "/project_folder/.venv/local/lib/python2.7/site-packages/ansiblelint/utils.py", line 518, in construct_mapping
    mapping = Constructor.construct_mapping(loader, node, deep=deep)
  File "/project_folder/.venv/local/lib/python2.7/site-packages/yaml/constructor.py", line 208, in construct_mapping
    return BaseConstructor.construct_mapping(self, node, deep=deep)
  File "/project_folder/.venv/local/lib/python2.7/site-packages/yaml/constructor.py", line 133, in construct_mapping
    value = self.construct_object(value_node, deep=deep)
  File "/project_folder/.venv/local/lib/python2.7/site-packages/yaml/constructor.py", line 88, in construct_object
    data = constructor(self, node)
  File "/project_folder/.venv/local/lib/python2.7/site-packages/yaml/constructor.py", line 414, in construct_undefined
    node.start_mark)
yaml.constructor.ConstructorError: could not determine a constructor for the tag '!unsafe'
  in "<unicode string>", line 18, column 14:
        inspect: !unsafe '{{.State.Running}}'
```
- if you're getting a stack trace, output of
  `ansible-playbook --syntax-check playbook` 
This yielded no results.

**Comment by willthames** [comment]
The Ansible YAML parsing has moved on a long way since we added linenumber parsing to ansible-lint. 

Looks like ansible-lint might need some refactoring to better reuse some of those changes - this is not a trivial fix.

**Comment by rhpvorderman** [comment]
Would it be possible to use [ansible's own constructor](https://github.com/ansible/ansible/blob/c38ff3b8f84aed46fd087d91a457a6372eda9609/lib/ansible/parsing/yaml/constructor.py)?
```python
from ansible.parsing.yaml import AnsibleContructor as Constructor
```
Or something like that.

**Comment by willthames** [comment]
Yeah, that was the first thing I tried, and it quickly took me down a rabbit hole as I'd need to change to the AnsibleLoader, and then I don't know how I'd achieve what is currently supposed to happen (obtaining linenumbers) but there is a possibility it might come for free with moving over to Ansible's YAML parsing - I just don't have time to look into this in as much depth as is needed right now, but I acknowledge it's definitely a bug!

**Comment by willthames** [comment]
This seems to be fixed in master now, probably through #332. 

Using 3.5.0rc1, I get: 
```
$ ansible-lint play.yml 
[ANSIBLE0012] Commands should not change things if nothing needs doing
play.yml:4
Task/Handler: register whether the container has finished
```

**Comment by rhpvorderman** [comment]
Thanks for fixing this. The pull request looks very nice!

**Comment by willthames** [comment]
Fixed in 3.4.22, now available through pip
